src/core/lib/iomgr/port.h

```diff
@@ -140,10 +138,8 @@
 #error "Platform not recognized"
 #endif
 
-#if defined(GRPC_POSIX_SOCKET) + defined(GRPC_WINSOCK_SOCKET) + \
-        defined(GRPC_CUSTOM_SOCKET) + defined(GRPC_UV) !=       \
-    1
-#error Must define exactly one of GRPC_POSIX_SOCKET, GRPC_WINSOCK_SOCKET, GPR_CUSTOM_SOCKET
+#if defined(GRPC_POSIX_SOCKET) + defined(GRPC_WINSOCK_SOCKET) != 1
+#error Must define exactly one of GRPC_POSIX_SOCKET, GRPC_WINSOCK_SOCKET
```

Comment:
All the stuff that gets abstracted has a ```grpc_default_<pollset|pollset_set|tcp_server>_vtable``` that points to either the windows or  posix implementation, and that will get used if we don't specify overrides.

If we want to avoid the requirement that ```GRPC_POSIX_SOCKET``` or ```GRPC_WINSOCK_SOCKET``` is defined, we could provide another empty implementation that just asserts() if the application attempts to use the default without overriding anything..